thrift/thrift-gen/compile_test.go

```diff
@@ -30,14 +30,71 @@ import (
 	"path"
 	"path/filepath"
 	"strings"
+	"sync"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go/testutils"
 )
 
 // These tests ensure that the code generator generates valid code that can be built
 // in combination with Thrift's autogenerated code.
 
+var (
+	_testGoPath     string
+	_testGoPathOnce sync.Once
+)
+
+func TestMain(m *testing.M) {
+	exitCode := m.Run()
+
+	// If we created a fake GOPATH, we should clean it up.
+	if _testGoPath != "" {
+		os.RemoveAll(_testGoPath)
+	}
+
+	os.Exit(exitCode)
+}
+
+func getTChannelDir(goPath string) string {
+	return filepath.Join(goPath, "src", "github.com", "uber", "tchannel-go")
+}
+
+func createGoPath(t *testing.T) {
+	goPath, err := ioutil.TempDir("", "thrift-gen")
+	require.NoError(t, err, "TempDir failed")
+
+	// Create $GOPATH/src/github.com/uber/tchannel-go and symlink everything.
+	// And then create a dummy directory for all the test output.
+	tchannelDir := getTChannelDir(goPath)
+	require.NoError(t, os.MkdirAll(tchannelDir, 0755), "MkDirAll failed")
+
+	// Symlink everything in the real GOPATH.
+	realTChannelDir := getTChannelDir(os.Getenv("GOPATH"))
```

Comment:
What about when `GOPATH="$path1:$path2"`? Also, aren't tests run from inside
the directory containing the test file? Is it too hacky to walk up the
directory tree and find the tchannel-go directory?